### DISCUSSION TITLE
Integrating socket.io(^3.0.0) with strapi("3.2.5") cors error

### DISCUSSION THREAD

**Discussion by purnimagupta** [strapi/strapi#8593] [Legacy Topics]
I'm trying to build a simple chat app. I'm using create-react-app for the front-end, and I'm trying to integrate socket.io with Strapi server. My create-react-app server is running on Port 3000, while my Strapi local server is running on port 1337. 

I've used this command to install starpi `yarn create strapi-app chat-app` and manually selected `mongodb` as database.

Receiving this error in the React app:

![react-socket-cors-err](https://user-images.githubusercontent.com/4417175/98506063-b3a49c80-2280-11eb-9e3f-0770254c16a0.png)

# Code
On my React component, I have this:

```
import io from 'socket.io-client';
const socket = io('http://localhost:1337');

 const handleClick = () => {        
        const username = props.username;
        socket.emit("sendMessage", {
            chatMessage,
            username
        })
    }
```

In the Strapi backend, inside `./config/functions/bootstrap.js`

```
module.exports = async () => {
    process.nextTick(() => {
        var io = require('socket.io')(strapi.server);

        strapi.io.on('connection', async function(socket) {

            socket.on('sendMessage', (msg) => {
                console.log("msg is", msg)
            })
            socket.on('disconnect', () =>{
                console.log('a user disconnected')
            });
        });
         strapi.io = io;
      })
};
```

I know to solve this `cors` error inside node app using cors module and use it like `app.use(cors())` But not sure how I resolve this in case of strapi?

I followed [this thread](https://github.com/strapi/strapi/issues/5869) and tried some of the workarounds, but none of them seem to work. How do I resolve this issue?

**Comment by purnimagupta** [comment]
I resolve the issue by passing cors object to socket.io. Options `origins` has been replaced by `cors` in the latest version of socket.io(^3.0.0)

```
  var io = require('socket.io')(strapi.server, {
            cors: {
              origin: "http://localhost:3000",
              methods: ["GET", "POST"],
              allowedHeaders: ["my-custom-header"],
              credentials: true
            }
        });
```

It works now :)

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Reply by purnimagupta** [reply]
Sure, Should I paste the question there as well along with my answer?

**Reply by derrickmehaffy** [reply]
Please do yes just so we don't lose any useful information when we disable the discussions
